src/app/public/src/matchers/matchers.ts

```diff
@@ -0,0 +1,173 @@
+          'Expected element to be visible';
+
+        return result;
+      }
+    };
+  },
+
+  toHaveText() {
+    return {
+      compare(el: any, expectedText: string, trimWhitespace = true): jasmine.CustomMatcherResult {
+        const result = {
+          pass: false,
+          message: ''
+        };
+
+        let actualText = el.textContent;
+
+        if (trimWhitespace) {
+          actualText = actualText.trim();
+        }
+
+        result.pass = actualText === expectedText;
+
+        result.message = result.pass ?
+          `Expected element's inner text not to be ${expectedText}` :
+          `Expected element's inner text to be: "${expectedText}"\n` +
+          `Actual element's inner text was: "${actualText}"`;
+
+        return result;
+      }
+    };
+  },
+
+  toHaveCssClass() {
+    return {
+      compare(el: any, expectedClassName: string): jasmine.CustomMatcherResult {
+        const result = {
+          pass: false,
+          message: ''
+        };
+
+        if (expectedClassName.indexOf('.') === 0) {
+          throw new Error('Please remove the leading dot from your class name.');
+        }
+
+        result.pass = el.classList.contains(expectedClassName);
+
+        result.message = result.pass ?
+          `Expected element not to have CSS class "${expectedClassName}"` :
+          `Expected element to have CSS class "${expectedClassName}"`;
+
+        return result;
+      }
+    };
+  },
+
+  toHaveStyle() {
+    return {
+      compare(el: any, expectedStyles: any): jasmine.CustomMatcherResult {
+        const message: string[] = [];
+
+        let hasFailure = false;
+
+        Object.keys(expectedStyles).forEach((styleName: string) => {
+          const styles = windowRef.getComputedStyle(el);
+          const actualStyle = styles[styleName];
+          const expectedStyle = expectedStyles[styleName];
+
+          if (actualStyle !== expectedStyle) {
+            if (!hasFailure) {
+              hasFailure = true;
+            }
+
+            message.push(
+              `Expected element not to have CSS style "${styleName}: ${expectedStyle}"`
+            );
+          } else {
+            message.push(
+              `Expected element to have CSS style "${styleName}: ${expectedStyle}"`
+            );
+          }
+
+          message.push(
+            `Actual styles are: "${styleName}: ${actualStyle}"`
+          );
+        });
+
+        const result = {
+          pass: !hasFailure,
+          message: message.join('\n')
+        };
+
+        return result;
+      }
+    };
+  },
+
+  toExist() {
+    return {
+      compare(el: any): jasmine.CustomMatcherResult {
+        const result = {
+          pass: false,
+          message: ''
+        };
+
+        result.pass = !!el;
+
+        result.message = result.pass ?
+          'Expected element not to exist' :
+          'Expected element to exist';
+
+        return result;
+      }
+    };
+  },
+
+  toPassA11y() {
```

Comment:
I'd change the name of this function to `toBeAccessible()`.  It's longer but is more in line with some of the other built-in matchers (`toBeNull`, `toBeUndefined()`, `toBeTruthy()`, `toBeGreaterThan()`, etc.).